In [19]:
import pandas as pd
import numpy as np
import os
import re
from langdetect import detect
from phonemizer import phonemize
from unidecode import unidecode

text = 'hello'
result = unidecode(text)
print(result)

hello


In [15]:
phonemize("text", language="en-us", strip=True, preserve_punctuation=True, with_stress=True, njobs=1, language_switch="remove-flags")

RuntimeError: espeak not installed on your system

In [ ]:
# find tiktok_videos -type f -name "*.mp4" -exec sh -c 'mkdir -p tiktok_audio/"$(dirname "{}")" && ffmpeg -i "{}" -vn tiktok_audio/"{}".wav' \;

In [2]:
root_dirs = ['../data/tiktok_videos/test' ,  '../data/tiktok_videos/val' , '../data/tiktok_videos/train']
subfolders = ['educative', 'others', 'suggestive']
data = []
dc = {'others' : 0 , 'suggestive' : 1 , 'educative' : 2}
for root_dir in root_dirs:
    for subfolder in subfolders:
        path = os.path.join(root_dir, subfolder)
        for root, dirs, files in os.walk(path):
            for file in files:
                if file.endswith('.mp4'):
                    full_path = os.path.join(root, file)
                    text_path = "../data/transcripts/" + full_path.split('../data/tiktok_videos/')[1][:-4] + ".txt"
                    with open(text_path, 'r') as f:
                        text = f.read()
                        data.append([ re.sub(r"_\d+", "", full_path.split('/')[-1][:-4]), subfolder , dc[subfolder] , text , full_path , 
                                     full_path.replace("tiktok_videos" , "tiktok_wav")+".wav" , 
                                     root_dir.split('/')[-1] , len(text)])
                        
# text , audio_path , video_path , label , label_name , split , dialog , timings

df = pd.DataFrame(data, columns=['name' , 'content_label', 'content'  , 'text' , 'video_path' , 'audio_path', 'split' , 'text_len'])
df.loc[df['text_len'] == 0, "text"] = "[N/A]"

df_train = df[df['split'] == 'train']
df_test = df[df['split'] == 'test']
df_val = df[df['split'] == 'val']
df_train['dialog'] = df_train.groupby('name').ngroup()
df_test['dialog'] = df_test.groupby('name').ngroup()
df_val['dialog'] = df_val.groupby('name').ngroup()

df = pd.concat([df_test , df_val , df_train])

df

/home/jupyter/new_venv/lib/python3.7/site-packages/ipykernel_launcher.py:27: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
/home/jupyter/new_venv/lib/python3.7/site-packages/ipykernel_launcher.py:28: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
/home/jupyter/new_venv/lib/python3.7/site-packages/ipykernel_launcher.py:29: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https:

name content_label  content  \
0      alexandereadie     educative        2   
1            viv_whnp     educative        2   
2          leanna.apa     educative        2   
3     mel.on.medicine     educative        2   
4           contro_za     educative        2   
..                ...           ...      ...   
995       meowbarbie5    suggestive        1   
996        mich.bunny    suggestive        1   
997    ashleyhandal00    suggestive        1   
998  misterfelipemaia    suggestive        1   
999  kendra.the.nurse    suggestive        1   

                                                  text  \
0     Take my mind and take my pain Like an empty b...   
1                                                [N/A]   
2     Hello, you are here for your very first pap s...   
3     You knew I had to talk about this article too...   
4     Facts nobody told you about your penis. Men w...   
..                                                 ...   
995              Make it, make it, make that disco pop   
996                          Where it be? Where it be?   
997   Tu eres una yala y tu eres mala baby No te va...   
998   Why, why me? Let the night, let the night, le...   
999   I'm fresh as dinner, motherfucker Okay Okay I...   

                                            video_path  \
0    ../data/tiktok_videos/test/educative/alexander...   
1    ../data/tiktok_videos/test/educative/viv_whnp_...   
2    ../data/tiktok_videos/test/educative/leanna.ap...   
3    ../data/tiktok_videos/test/educative/mel.on.me...   
4    ../data/tiktok_videos/test/educative/contro_za...   
..                                                 ...   
995  ../data/tiktok_videos/train/suggestive/meowbar...   
996  ../data/tiktok_videos/train/suggestive/mich.bu...   
997  ../data/tiktok_videos/train/suggestive/ashleyh...   
998  ../data/tiktok_videos/train/suggestive/misterf...   
999  ../data/tiktok_videos/train/suggestive/kendra....   

                                            audio_path  split  text_len  \
0    ../data/tiktok_wav/test/educative/alexanderead...   test        66   
1    ../data/tiktok_wav/test/educative/viv_whnp_681...   test         0   
2    ../data/tiktok_wav/test/educative/leanna.apa_6...   test      1104   
3    ../data/tiktok_wav/test/educative/mel.on.medic...   test      1700   
4    ../data/tiktok_wav/test/educative/contro_za_71...   test       192   
..                                                 ...    ...       ...   
995  ../data/tiktok_wav/train/suggestive/meowbarbie...  train        38   
996  ../data/tiktok_wav/train/suggestive/mich.bunny...  train        26   
997  ../data/tiktok_wav/train/suggestive/ashleyhand...  train        96   
998  ../data/tiktok_wav/train/suggestive/misterfeli...  train        80   
999  ../data/tiktok_wav/train/suggestive/kendra.the...  train        59   

     dialog  
0        19  
1       179  
2       153  
3       157  
4        88  
..      ...  
995     219  
996     222  
997      10  
998     233  
999     140  

[1000 rows x 9 columns]

In [5]:
def fn(text):
    try:
        return detect(text)
    except:
        return None


df['language'] = df['text'].apply(fn)

name content_label  content  \
0      alexandereadie     educative        2   
2          leanna.apa     educative        2   
3     mel.on.medicine     educative        2   
4           contro_za     educative        2   
6    intimacywithjess     educative        2   
..                ...           ...      ...   
993    ashleyhandal00    suggestive        1   
995       meowbarbie5    suggestive        1   
996        mich.bunny    suggestive        1   
998  misterfelipemaia    suggestive        1   
999  kendra.the.nurse    suggestive        1   

                                                  text  \
0     Take my mind and take my pain Like an empty b...   
2     Hello, you are here for your very first pap s...   
3     You knew I had to talk about this article too...   
4     Facts nobody told you about your penis. Men w...   
6     This video is for educational purposes only. ...   
..                                                 ...   
993                      Kitano wins flawless victory.   
995              Make it, make it, make that disco pop   
996                          Where it be? Where it be?   
998   Why, why me? Let the night, let the night, le...   
999   I'm fresh as dinner, motherfucker Okay Okay I...   

                                            video_path  \
0    ../data/tiktok_videos/test/educative/alexander...   
2    ../data/tiktok_videos/test/educative/leanna.ap...   
3    ../data/tiktok_videos/test/educative/mel.on.me...   
4    ../data/tiktok_videos/test/educative/contro_za...   
6    ../data/tiktok_videos/test/educative/intimacyw...   
..                                                 ...   
993  ../data/tiktok_videos/train/suggestive/ashleyh...   
995  ../data/tiktok_videos/train/suggestive/meowbar...   
996  ../data/tiktok_videos/train/suggestive/mich.bu...   
998  ../data/tiktok_videos/train/suggestive/misterf...   
999  ../data/tiktok_videos/train/suggestive/kendra....   

                                            audio_path  split  text_len  \
0    ../data/tiktok_wav/test/educative/alexanderead...   test        66   
2    ../data/tiktok_wav/test/educative/leanna.apa_6...   test      1104   
3    ../data/tiktok_wav/test/educative/mel.on.medic...   test      1700   
4    ../data/tiktok_wav/test/educative/contro_za_71...   test       192   
6    ../data/tiktok_wav/test/educative/intimacywith...   test       946   
..                                                 ...    ...       ...   
993  ../data/tiktok_wav/train/suggestive/ashleyhand...  train        30   
995  ../data/tiktok_wav/train/suggestive/meowbarbie...  train        38   
996  ../data/tiktok_wav/train/suggestive/mich.bunny...  train        26   
998  ../data/tiktok_wav/train/suggestive/misterfeli...  train        80   
999  ../data/tiktok_wav/train/suggestive/kendra.the...  train        59   

     dialog language  
0        19       en  
2       153       en  
3       157       en  
4        88       en  
6       140       en  
..      ...      ...  
993      10       en  
995     219       en  
996     222       en  
998     233       en  
999     140       en  

[752 rows x 10 columns]

In [11]:
df[~(df['language'].isna()) & ~(df['language'] == 'en')]


name content_label  content  \
1          viv_whnp     educative        2   
5         dickdocsg     educative        2   
26      midwifemama     educative        2   
40      deborahyowa        others        0   
51         boxlapse        others        0   
..              ...           ...      ...   
987   imonlyevilena    suggestive        1   
988            okhp    suggestive        1   
992   taniarosmeris    suggestive        1   
994    starmuscle69    suggestive        1   
997  ashleyhandal00    suggestive        1   

                                                  text  \
1                                                [N/A]   
5     Hey guys, jika anda melihat pembuatan putih d...   
26    I'm taking a chance, cause I like you a lot, ...   
40   笑 snap .本当に美味しい,ちちゃくちゃ美味しい楽しんだ期待している今回のメニューはあと...   
51                                               [N/A]   
..                                                 ...   
987   te sale este video en para ti? llegaste al vi...   
988                                              [N/A]   
992   demonios tiene tu cuerpo lleno de tinta y ese...   
994                                                you   
997   Tu eres una yala y tu eres mala baby No te va...   

                                            video_path  \
1    ../data/tiktok_videos/test/educative/viv_whnp_...   
5    ../data/tiktok_videos/test/educative/dickdocsg...   
26   ../data/tiktok_videos/test/educative/midwifema...   
40   ../data/tiktok_videos/test/others/deborahyowa_...   
51   ../data/tiktok_videos/test/others/boxlapse_699...   
..                                                 ...   
987  ../data/tiktok_videos/train/suggestive/imonlye...   
988  ../data/tiktok_videos/train/suggestive/okhp_71...   
992  ../data/tiktok_videos/train/suggestive/taniaro...   
994  ../data/tiktok_videos/train/suggestive/starmus...   
997  ../data/tiktok_videos/train/suggestive/ashleyh...   

                                            audio_path  split  text_len  \
1    ../data/tiktok_wav/test/educative/viv_whnp_681...   test         0   
5    ../data/tiktok_wav/test/educative/dickdocsg_70...   test       223   
26   ../data/tiktok_wav/test/educative/midwifemama_...   test       230   
40   ../data/tiktok_wav/test/others/deborahyowa_716...   test        62   
51   ../data/tiktok_wav/test/others/boxlapse_699777...   test         0   
..                                                 ...    ...       ...   
987  ../data/tiktok_wav/train/suggestive/imonlyevil...  train       129   
988  ../data/tiktok_wav/train/suggestive/okhp_71474...  train         0   
992  ../data/tiktok_wav/train/suggestive/taniarosme...  train       104   
994  ../data/tiktok_wav/train/suggestive/starmuscle...  train         4   
997  ../data/tiktok_wav/train/suggestive/ashleyhand...  train        96   

     dialog language  
1       179       hu  
5        98       id  
26      158       it  
40       97       ja  
51       60       hu  
..      ...      ...  
987      92       es  
988     282       hu  
992     399       es  
994     386       sw  
997      10       es  

[241 rows x 10 columns]

In [12]:
df['language'].value_counts()

en       752
hu        69
es        30
id        17
ko        14
pt        12
sq        11
sw        11
fr        10
tl        10
cy        10
de         9
it         8
sv         4
ja         4
tr         4
so         4
af         3
ru         3
zh-cn      2
hi         1
bg         1
nl         1
ar         1
no         1
cs         1
Name: language, dtype: int64

At least 752 data points are english, 6 have no language, one is just numbers. So 241 are actually different languages.


In [97]:
#set pandas dataframe to full width
pd.set_option('display.max_rows', 500)
pd.set_option('display.max_columns', 500)
pd.set_option('display.max_colwidth',1000)
pd.set_option('display.width', 1000)
df[['audio_path' , 'video_path']].loc[0]

audio_path    ../data/tiktok_wav/test/educative/alexandereadie_6955257178760547590.mp4.wav
video_path     ../data/tiktok_videos/test/educative/alexandereadie_6955257178760547590.mp4
Name: 0, dtype: object

In [101]:
df.to_pickle(f"../data/tiktok.pkl")

Median number of words is 29 , average is 85. We are going to do 300 since that is what we have always done